package.json

```diff
@@ -33,6 +33,7 @@
     "consola": "^3.2.3",
     "defu": "^6.1.4",
     "h3": "^1.11.1",
+    "hash.js": "^1.1.7",
```

Comment:
as you've suggested, let's switch to `ohash` 🙏

Reply:
I'm not so experienced in hash or ohash, but I think that ohash misses the `hmac` function to generate the signature. 

```ts
  const hmac = hash.hmac(hash.sha256, hexDecode(secret));
```

More in general, I'm open to suggestions on this topic.

Reply:
I have a implement at https://github.com/nuxt/image/issues/963, using [`uncrypto`](https://github.com/unjs/uncrypto)

Reply:
Ty for the hint, I will work on it tomorrow or later today👍🏻

Reply:
@everyx 
I see, .. so to properly support imgproxy, and to do it safely or imgproxy will allow us to optimize any URL provided; we are waiting for:
https://github.com/nuxt/image/issues/276 --> To securely sign the URLs with uncrypto. 
https://github.com/nuxt/image/issues/963 --> To support getImage and not just the NuxtImg component. 
I hope to get it right,

For now, as an alternative, @danielroe, we could remove the signing of the URL and add big, pretty big, I would say, disclaimer to use the EnvVar `IMGPROXY_ALLOWED_SOURCES` to secure the install of imgproxy.